nixos/modules/services/games/minecraft-server.nix

```diff
@@ -15,7 +16,23 @@ in
           If enabled, start a Minecraft Server. The listening port for
           the server is always <literal>25565</literal>. The server
           data will be loaded from and saved to
-          <literal>/var/lib/minecraft</literal>.
+          <literal>${cfg.dataDir}</literal>.
+        '';
+      };
+
+      dataDir = mkOption {
+        type = types.path;
+        default = "/var/lib/minecraft";
+        description = ''
+          Directory to store minecraft database and other state/data files.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = true;
```

Comment:
The consensus is not to enable ports by default. You can add documentation to `enable` option what is needed for service to function correctly (and provide an example)